### PR TITLE
[SPARK-17686][Core] Support printing out scala and java version with spark-submit --version command

### PR DESCRIPTION
**Pull request by jerryshao**
## What changes were proposed in this pull request?

In our universal gateway service we need to specify different jars to Spark according to scala version. For now only after launching Spark application can we know which version of Scala it depends on. It makes hard for us to support different Scala + Spark versions to pick the right jars.

So here propose to print out Scala version according to Spark version in "spark-submit --version", so that user could leverage this output to make the choice without needing to launching application.
## How was this patch tested?

Manually verified in local environment.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala [hunk 1]
```diff
@@ -104,6 +104,8 @@ object SparkSubmit {
    /___/ .__/\_,_/_/ /_/\_\   version %s
       /_/
                         """.format(SPARK_VERSION))
+    printStream.println("Using Scala %s (%s, Java %s)".format(
```
Comment (rxin) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala]: How about just ...

"Using Scala %s, %s, %s"

Comment (rxin) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala]: Basically Properties.javaVmName already says "Java".

Comment (jerryshao) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala]: Thanks Reynold for your comments. I will change it.

**Comment by rxin** [comment]
LGTM pending Jenkins.

**Comment by andrewor14** [comment]
Merging into master

**Comment by jerryshao** [comment]
Thanks @rxin and @andrewor14 .

**Comment by jerryshao** [comment]
Looks like unrelated test failure, it can be passed in my local test.

**Pull request closed without merging**